### PR TITLE
feat(ospf): implement OSPF generator for Phase 3.2 and 3.3

### PR DESCRIPTION
**Pull request by duplico**
## Summary

Implements OSPF generator for VyOS Sagitta using interface-based configuration syntax.

## Phase 3.2 - OSPF Basic

- Interface-based area assignment: `set protocols ospf interface ethX area Y`
- Passive interfaces: `set protocols ospf interface ethX passive`
- Cost overrides: `set protocols ospf interface ethX cost N`
- Cross-reference validation (interfaces must exist)

## Phase 3.3 - OSPF Advanced

- Router ID: `set protocols ospf parameters router-id X.X.X.X`
- Redistribution: `set protocols ospf redistribute connected|static|kernel`
- Default-information originate with `always` and `metric` options

## Design Decisions

- **Sagitta syntax**: Uses `interface X area Y` (not legacy `area X network Y`)
- **Data-plane only**: OSPF never configured in management VRF
- **No authentication**: Matches equuleus behavior
- **Explicit areas**: All interfaces require explicit area assignment

## Test Coverage

- 17 new unit tests covering all OSPF functionality
- Integration test context validates OSPF command generation
- Edge cases: None config, disabled, minimal, full-featured

## Test plan

- [x] `just check` passes (ruff, mypy, pytest)
- [x] CI passes
- [x] Integration test with OSPF context

Generated with Claude Code (Opus 4.5)

### CODE REVIEW

File: tests/test_generators.py [hunk 1]
```diff
@@ -1116,3 +1124,318 @@ def test_generate_config_command_order_with_vrf(self):
...
+            interfaces=[
+                InterfaceConfig(
+                    name="eth0",
+                    ip=IPv4Address("10.0.1.1"),
+                    mask="255.255.255.0",
+                ),
+                InterfaceConfig(
+                    name="eth1",
+                    ip=IPv4Address("10.64.1.1"),
+                    mask="255.255.255.0"),
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: really? is that what the codebase has? that sounds ugly?

File: SUMMARY.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+The implementation is ready for PR creation and review.
+
+---
+
+**Implementation completed by**: Claude Opus 4.5 (via Claude Code)
+**Date**: 2026-01-19
+**Worktree**: `/home/george/swccdc/vyos-onecontext-worktrees/ospf-1737336000`
+**Branch**: `feat/ospf-1737336000`
+**Commit**: `1570291`
```
[the remark this answers is not in the document]
Comment (duplico) [reply SUMMARY.md]: Fixed in commit f13c156. SUMMARY.md has been removed.

(AI-generated via Claude Code w/ Sonnet 4.5)

File: tests/test_generators.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: Fixed in commit f13c156. The closing parenthesis is now on its own line to match the formatting pattern used throughout the file.

(AI-generated via Claude Code w/ Sonnet 4.5)

File: tests/test_generators.py [hunk 3]
```diff
@@ -1116,3 +1124,318 @@ def test_generate_config_command_order_with_vrf(self):
...
+                    name="eth0",
+                    ip=IPv4Address("10.0.1.1"),
+                    mask="255.255.255.0",
+                    gateway=IPv4Address("10.0.1.254"),
+                    management=True,
+                ),
+                InterfaceConfig(
+                    name="eth1",
+                    ip=IPv4Address("10.64.1.1"),
+                    mask="255.255.255.0"),
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: Fixed in commit f13c156. The closing parenthesis is now on its own line to match the formatting pattern used throughout the file.

(AI-generated via Claude Code w/ Sonnet 4.5)

File: tests/test_generators.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: You're right that the Copilot wording was confusing - it's not about adding a space before the closing paren. The actual issue is that the closing paren should be on its own line to match the formatting pattern used throughout the rest of the file. Compare:

Wrong (lines 1367, 1417):
```python
InterfaceConfig(
    name="eth1",
    ip=IPv4Address("10.64.1.1"),
    mask="255.255.255.0"),
```

Right (everywhere else in the file):
```python
InterfaceConfig(
    name="eth1",
    ip=IPv4Address("10.64.1.1"),
    mask="255.255.255.0",
),
```

Fixed in f13c156.

(AI-generated via Claude Code w/ Sonnet 4.5)

**Comment by duplico** [comment]
Fixed the duplicate `assert_command_generated` function in commit 2201fd4. Removed the fixed-string version (grep -qF) and kept the regex-capable version needed for pattern matching like `.*192.168.122.30`.

**Comment by duplico** [comment]
Fixed the missing blank line before `TestOspfGenerator` in commit b795ad3.

Regarding the `__main__.py` formatting observation: Acknowledged. The log message reformatting was inherited from an earlier commit in the rebase chain (the test validation fix). Keeping it as-is since separating it out would complicate the commit history, but will be more careful about isolating formatting changes in future PRs.

**Comment by duplico** [comment]
Fixed the misleading 'Management interface' comment in ospf.env (commit 68dba49).

The multi-interface test failure appears to be a flaky CI issue - this same test passed on the sagitta branch when PR #71 was merged (run 21157714103). The error was 'VyOS wrapper command failed: commit / Unknown error' which suggests a timing/resource issue rather than a code problem. No code changes in this PR affect the multi-interface or alias handling. Re-running CI to verify.

**Comment by duplico** [comment]
Opened #73 to track the flaky multi-interface test issue.

File: tests/test_generators.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (duplico) [reply tests/test_generators.py]: oh, good, so the way it should be :-)
